<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1005</width>
    <height>866</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>330</y>
      <width>401</width>
      <height>481</height>
     </rect>
    </property>
    <property name="title">
     <string>MAC table</string>
    </property>
    <widget class="QPushButton" name="cleartableButton">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>50</y>
       <width>99</width>
       <height>27</height>
      </rect>
     </property>
     <property name="text">
      <string>clear</string>
     </property>
    </widget>
    <widget class="QComboBox" name="modifyTimerComboBox">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>50</y>
       <width>85</width>
       <height>27</height>
      </rect>
     </property>
     <property name="editable">
      <bool>true</bool>
     </property>
     <property name="sizeAdjustPolicy">
      <enum>QComboBox::AdjustToContentsOnFirstShow</enum>
     </property>
     <item>
      <property name="text">
       <string>1 sec</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>2 sec</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>3 sec</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>4 sec</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>5 sec</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>6 sec</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>7 sec</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>8 sec</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>9 sec</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>10 sec</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>entry timer</string>
     </property>
    </widget>
    <widget class="QListView" name="mactableListView">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>110</y>
       <width>361</width>
       <height>351</height>
      </rect>
     </property>
     <property name="lineWidth">
      <number>1</number>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAsNeeded</enum>
     </property>
     <property name="viewMode">
      <enum>QListView::ListMode</enum>
     </property>
     <property name="modelColumn">
      <number>0</number>
     </property>
     <property name="uniformItemSizes">
      <bool>false</bool>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>20</y>
      <width>401</width>
      <height>201</height>
     </rect>
    </property>
    <property name="title">
     <string>Interfaces</string>
    </property>
    <widget class="QComboBox" name="port1ComboBox">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>40</y>
       <width>231</width>
       <height>27</height>
      </rect>
     </property>
    </widget>
    <widget class="QComboBox" name="port2ComboBox">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>80</y>
       <width>231</width>
       <height>27</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>40</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>switch port 1:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>80</y>
       <width>111</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>switch port 2:</string>
     </property>
    </widget>
    <widget class="QPushButton" name="startButton">
     <property name="geometry">
      <rect>
       <x>248</x>
       <y>150</y>
       <width>131</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Start switching</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>20</y>
      <width>251</width>
      <height>421</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="title">
     <string>Port 1 Stats</string>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>30</y>
       <width>21</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>IN</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>30</y>
       <width>31</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>OUT</string>
     </property>
    </widget>
    <widget class="QLCDNumber" name="ipCountIN">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>140</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="eth2CountOUT">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>90</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="ipCountOUT">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>140</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="arpCountOUT">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>190</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="arpCountIN">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>190</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="tcpCountIN">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>240</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="eth2CountIN">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>90</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="udpCountOUT">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>290</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="udpCountIN">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>290</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="tcpCountOUT">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>240</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>71</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Ethernet II</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>68</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>IP</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>190</y>
       <width>68</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>ARP</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_14">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>240</y>
       <width>68</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>TCP</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_15">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>290</y>
       <width>68</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>UDP</string>
     </property>
    </widget>
    <widget class="QLCDNumber" name="lcdNumber_22">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>310</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="lcdNumber_23">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>310</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="icmpCountIN">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>340</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="icmpCountOUT">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>340</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_16">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>340</y>
       <width>68</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>ICMP</string>
     </property>
    </widget>
    <widget class="QLCDNumber" name="httpCountIN">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>390</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="httpCountOUT">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>390</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_17">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>390</y>
       <width>68</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>HTTP</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>740</x>
      <y>20</y>
      <width>251</width>
      <height>421</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <property name="title">
     <string>Port 2 Stats</string>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
    <widget class="QLabel" name="label_18">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>30</y>
       <width>21</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>IN</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_19">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>30</y>
       <width>31</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>OUT</string>
     </property>
    </widget>
    <widget class="QLCDNumber" name="ipCountIN_2">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>140</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="eth2CountOUT_2">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>90</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="ipCountOUT_2">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>140</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="arpCountOUT_2">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>190</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="arpCountIN_2">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>190</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="tcpCountIN_2">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>240</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="eth2CountIN_2">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>90</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="udpCountOUT_2">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>290</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="udpCountIN_2">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>290</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="tcpCountOUT_2">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>240</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_20">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>71</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Ethernet II</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_21">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>68</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>IP</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_22">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>190</y>
       <width>68</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>ARP</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_23">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>240</y>
       <width>68</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>TCP</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_24">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>290</y>
       <width>68</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>UDP</string>
     </property>
    </widget>
    <widget class="QLCDNumber" name="lcdNumber_24">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>310</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="lcdNumber_25">
     <property name="geometry">
      <rect>
       <x>350</x>
       <y>310</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="icmpCountIN_2">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>340</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="icmpCountOUT_2">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>340</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_25">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>340</y>
       <width>68</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>ICMP</string>
     </property>
    </widget>
    <widget class="QLCDNumber" name="httpCountIN_2">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>390</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLCDNumber" name="httpCountOUT_2">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>390</y>
       <width>41</width>
       <height>23</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_26">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>390</y>
       <width>68</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>HTTP</string>
     </property>
    </widget>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>30</y>
      <width>20</width>
      <height>421</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>30</y>
      <width>20</width>
      <height>661</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>270</y>
      <width>401</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>460</y>
      <width>541</width>
      <height>341</height>
     </rect>
    </property>
    <property name="whatsThis">
     <string extracomment="src addr"/>
    </property>
    <property name="title">
     <string>Filters</string>
    </property>
    <widget class="QListView" name="filterListView">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>130</y>
       <width>521</width>
       <height>161</height>
      </rect>
     </property>
    </widget>
    <widget class="QComboBox" name="filterpermitComboBox">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>71</width>
       <height>27</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>permit</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>deny</string>
      </property>
     </item>
    </widget>
    <widget class="QComboBox" name="filterportComboBox">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>30</y>
       <width>51</width>
       <height>27</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>1</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>2</string>
      </property>
     </item>
    </widget>
    <widget class="QComboBox" name="filterdirectionComboBox">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>30</y>
       <width>51</width>
       <height>27</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>in</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>out</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>any</string>
      </property>
     </item>
    </widget>
    <widget class="QComboBox" name="filterprotocolComboBox">
     <property name="geometry">
      <rect>
       <x>370</x>
       <y>30</y>
       <width>71</width>
       <height>27</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>HTTP</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>ICMP</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>UDP</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>TCP</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>IP</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>ARP</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>any</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_27">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>70</y>
       <width>81</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>HW address</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_28">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>100</y>
       <width>71</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>IP address</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="filter_hwsrc_Text">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>60</y>
       <width>141</width>
       <height>27</height>
      </rect>
     </property>
     <property name="toolTip">
      <string/>
     </property>
     <property name="whatsThis">
      <string/>
     </property>
     <property name="accessibleName">
      <string/>
     </property>
     <property name="inputMask">
      <string/>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="placeholderText">
      <string>source</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="filter_ipsrc_Text">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>90</y>
       <width>141</width>
       <height>27</height>
      </rect>
     </property>
     <property name="placeholderText">
      <string>source</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="filter_ipdst_Text">
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>90</y>
       <width>141</width>
       <height>27</height>
      </rect>
     </property>
     <property name="placeholderText">
      <string>destination</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="filter_hwdst_Text">
     <property name="geometry">
      <rect>
       <x>300</x>
       <y>60</y>
       <width>141</width>
       <height>27</height>
      </rect>
     </property>
     <property name="placeholderText">
      <string>destination</string>
     </property>
    </widget>
    <widget class="QPushButton" name="addFilterButton">
     <property name="geometry">
      <rect>
       <x>450</x>
       <y>86</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>ADD</string>
     </property>
    </widget>
    <widget class="QPushButton" name="deleteFilterButton">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>300</y>
       <width>99</width>
       <height>27</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>delete selected</string>
     </property>
     <property name="text">
      <string>delete</string>
     </property>
    </widget>
    <widget class="QPushButton" name="clearFiltersButton">
     <property name="geometry">
      <rect>
       <x>430</x>
       <y>300</y>
       <width>99</width>
       <height>27</height>
      </rect>
     </property>
     <property name="toolTip">
      <string>clear filters</string>
     </property>
     <property name="text">
      <string>clear</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_29">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>26</y>
       <width>41</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>port:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_30">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>30</y>
       <width>68</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>direction:</string>
     </property>
    </widget>
   </widget>
   <widget class="Line" name="line_4">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>440</y>
      <width>541</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1005</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
